mail_store_outgoing/models/ir_mail_server.py

```diff
@@ -0,0 +1,118 @@
+                res = {
+                    'server_id': self.id,
+                    'name': mailbox_name,
+                }
+                imap_pool.create(res)
+            self.write({'imap_mailbox_verified': True})
+        except Exception, e:
+            raise ValidationError(
+                _("Connection Test Failed!"),
+                _("Here is what we got instead:\n %s") % tools.ustr(e))
+        finally:
+            try:
+                if result:
+                    maillib.logout()
+            except Exception:
+                # ignored, just a consequence of the previous exception
+                pass
+
+    @api.model
+    def send_email(self, message, mail_server_id=None,
+                   smtp_server=None, smtp_port=None, smtp_user=None,
+                   smtp_password=None, smtp_encryption=None, smtp_debug=False):
+        super(IrMailServer, self).send_email(
+            message, mail_server_id, smtp_server, smtp_port,
+            smtp_user, smtp_password, smtp_encryption, smtp_debug)
+        return self._save_sent_message_to_sentbox(message, mail_server_id)
+
+    @api.model
+    def _save_sent_message_to_sentbox(self, msg, mail_server_id,):
+        smtp_server = None
+        ir_mail_server = self.env['ir.mail_server']
+        mail_server = None
+        mail_server = ir_mail_server.sudo().search([], order='sequence',
+                                                   limit=1)
+        if mail_server:
+            if not mail_server.store_outgoing_mail:
+                return True
+            if mail_server.has_separate_imap_server:
+                smtp_server = mail_server.separate_imap_server
+            else:
+                smtp_server = mail_server.smtp_host
+            smtp_user = mail_server.smtp_user
+            smtp_password = mail_server.smtp_pass
+        if not smtp_server:
+            raise ValidationError(_(
+                "Please define at least one SMTP server, or provide the"
+                " SMTP parameters explicitly."))
+        maillib = imaplib.IMAP4_SSL(smtp_server)
+        try:
+            maillib.login(smtp_user, smtp_password)
+            folder = mail_server.imap_mailbox_folder.name.join('""')
+            maillib.append(folder, r'\Seen', None, str(msg))
+        except Exception, ex:
+                _logger.error(_(
+                    'Failed attaching mail via imap to server %s %s')
+                    % (ex, msg))
+        finally:
+            maillib.logout()
+        return True
+
+    imap_mailbox_folder = fields.Many2one('ir.mail.imap.folder',
+                                          'Imap Folders',)
+    imap_mailbox_verified = fields.Boolean('IMAP Connection Verified', )
+    store_outgoing_mail = fields.Boolean('Store Outgoing Mail', )
+    has_separate_imap_server = fields.Boolean('Use Separate Imap Server', )
+    separate_imap_server = fields.Char('Separate Imap Server', )
+    active = fields.Boolean('Active', )
```

Comment:
By convention, fields are on top of the class (not blocking for me)

Reply:
I moved them to them top now. but the convention is 'shit'. As soon as you have computed fields, your methods have to be declared before the fields.

Reply:
You can put the name of the compute/inverse/search function as a string instead.
If using a computed domain/context, you should still use a lambda or the method cannot be inherited, so in any case it's better to stick to the standard.